packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -720,6 +720,10 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             },
         );
 
+        if (this.context.baseSnapshot) {
```

Comment:
Feels like you do not need new method on interface, this can be additional argument in constructor / createRootSummarizerNodeWithGC()

Reply:
I agree with that, but I just did it to be consistent with the way it happens elsewhere. Usually it happens when we get the snapshot which isn't until we realize the data store, etc. so we decide during `loadBaseSummary` everywhere else.